clickhouse_options.go

```diff
@@ -168,6 +168,10 @@ type Options struct {
 	// to respond to a single Read call for bytes over the connection.
 	// Can be overridden with context.WithDeadline.
 	ReadTimeout time.Duration
+
+	// Set a custom transport for the http client.
+	// The default transport configured by the library is passed in as an argument.
+	TransportFunc func(*http.Transport) (http.RoundTripper, error)
```

Comment:
curious the rationale behind accepting `*http.Transport` type and returning `http.RoundTripper`?. 

Given the use case and name `TransportFunc`, I would prefer to keep it simple like `func(*http.Transport) *http.Transport`. Even we don't need additional `error` return I think?.

Reply:
We need to return a `http.RoundTripper` because of the reason described in the previous comment. We could also just pass in the transport as a `http.RoundTripper` so the signature becomes:
```go
TransportFunc func(http.RoundTripper) http.RoundTripper
```
But then you can't access the fields of the transport, making the function less powerful. It would however cover our usecase regardless so I don't have a strong opinion regarding having a `*http.Transport` vs `http.RoundTripper` as the input argument.

Regarding the return error I just thought it's practical to have the option to return an error. But I guess you can also just do all fallible operations outside of that function so that the return error is unncessary.